src/pytest_codspeed/instruments/walltime.py

```diff
@@ -114,17 +115,18 @@ def from_list(
 
 
 @dataclass
-class Benchmark:
-    name: str
-    uri: str
-
-    config: BenchmarkConfig
-    stats: BenchmarkStats
+class WalltimeBenchmark(Benchmark):
```

Comment:
is it essential to inherit here?

Reply:
That way we do not have to define the properties twice

Reply:
I meant It's weird to inherit from a metadata class

Reply:
Seen together, let's do composition instead of inheritance, by having a new field `benchmark_metadata: BenchmarkMetadata`